frontend/frontend.php

```diff
@@ -249,14 +249,17 @@ public function switch_blog( $new_blog_id, $prev_blog_id ) {
 			$restore_curlang = $this->curlang->slug; // To always remember the current language through blogs.
 		}
 
-		$lang = $this->model->get_language( $restore_curlang );
+		$lang          = $this->model->get_language( $restore_curlang );
 		$this->curlang = $lang ? $lang : $this->model->get_default_language();
+		if ( empty( $this->curlang ) ) {
+			return;
+		}
 
 		if ( isset( $this->static_pages ) ) {
 			$this->static_pages->init();
 		}
 
-		$this->load_strings_translations();
+		$this->load_strings_translations( $this->curlang->slug );
```

Comment:
```suggestion
		$this->load_strings_translations( $this->curlang->locale );
```
Shouldn't it be the locale instead of the slug?

Reply:
Yes, my mistake.

That said, it does cause an issue.
As the curlang is **English**, when you pass the `en_US` locale, the `load_strings_translations()` does the `get_language()` on this locale and it gets the **French** language. (because it's the first language with `en_US` in the cache). :eyes: